flutter/news/lib/features/frontpage/presentation/bloc/top_headlines_state.dart

```diff
@@ -0,0 +1,29 @@
+part of 'articles_cubit.dart';
+
+abstract class ArticlesState extends Equatable {
+  const ArticlesState();
+}
+
+class TopHeadlinesInitial extends ArticlesState {
+  @override
+  List<Object> get props => [];
+}
+
+class TopHeadlinesLoading extends ArticlesState {
+  @override
+  List<Object> get props => [];
+}
+
+class TopHeadlinesLoaded extends ArticlesState {
+  final List<Article> topHeadlines;
```

Comment:
Depending on how good you want to do layering on this application it might be worth avoiding mixing business logic from presentation logic. 

Article seems a type used on the business logic layer, while TopHeadlinesLoaded is an object that belongs to the presentation.

Reply:
What @gbasile means is that it's a good idea to do a switch here between the type that we use on business logic to represent an article; `Article` and something we can use on the presentation layer to represent an article which is to be displayed; something like `ArticleViewState` which then is defined according to what we are planning to present.

For example an `ArticleViewState` can have fields `HeaderViewState` (formed by a title), `ContentViewState`, formed by a `content:string`, etc. 

---

Also, @Diogo00dev , perhaps we don't need `initial`, `loading` and `loaded` and everything can be reduced to `loading` and `idle` (having idle the list of articles that has been already loaded, can be initially empty).

But you will discover this later when actually presenting stuff

Reply:
Now, when coming to decide if we use a `ViewState` we need to also see how much it's needed. Perhaps a lot of what an `Article` currently is representes what a potential `ArticleViewState` would be. 

If things doesn't change a lot then perhaps that `ViewState` layer is not required.

Reply:
My advice is, don't be too clever, by that I mean, don't do things you are not sure you need right now, but keep them in mind.